pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -92,9 +92,12 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 	if err != nil {
 		return err
 	}
-
-	agentVmlinuzFile := filepath.Join(bootArtifactsFullPath, fmt.Sprintf("agent.%s-vmlinuz", a.cpuArch))
-	kernelReader, err := os.Open(filepath.Join(a.tmpPath, "images", "pxeboot", "vmlinuz"))
+	kernelFileType := "vmlinuz"
+	if a.cpuArch == arch.RpmArch(types.ArchitectureS390X) {
+		kernelFileType = "kernel.img"
```

Comment:
nit: do you maybe know if this filename is configured somewhere (similar for the ignition file in the igninfo.json)? I didn't find anything around, but I've could have missed it

Reply:
Not sure @andfasano , we are only finding the kernel.img file under `images/pxeboot` for s390x and no vmlinuz so we tried to mitigate it like this

Reply:
Ack, thanks